src/forms/TimeInput.jsx

```diff
@@ -10,17 +44,136 @@ class TimeInput extends React.Component {
 	constructor(props) {
 		super(props);
 
+		this.state = {
+			supportsTime: true,
+			...this.parseValueIntoState(props.value, props.is24Hr)
+		};
+
+		this.onBlur = this.onBlur.bind(this);
+		this.onNumberChange = this.onNumberChange.bind(this);
+		this.onMeridianChange = this.onMeridianChange.bind(this);
+		this.onTimeInputChange = this.onTimeInputChange.bind(this);
 		this.onChange = this.onChange.bind(this);
 	}
 
 	/**
+	 * @description takes a time value and parses it out to be saved in state
+	 * @param {String} value - the new time value to be parsed for state
+	 * @param {Boolean} is24Hr - whether or not we are 24Hr time format
+	 */
+	parseValueIntoState(value, is24Hr) {
+		const timeParts = getTimeParts(value);
+		return {
+			[HOURS]: value && (formatDigits(timeParts[HOURS] % (is24Hr ? 24 : 12)) || '12'),
+			[MINUTES]: value && (formatDigits(timeParts[MINUTES]) || '00'),
+			[MERIDIAN]: !is24Hr && timeParts[MERIDIAN],
+			value
+		};
+	}
+
+	/**
 	* @function onChange
+	* @param partialState the information to update state with
+	* @description called when the hour or minute input loses focus, or when the browser-native
+	* 	time input chages. In turn calls the onChange handler prop, if there is one provided
+	* 	(eg supplied by redux-form or DateTimePicker) with the updated values
+	*/
+	onChange(partialState) {
+		if (this.props.onChange) {
+			const stateValues = {
+				...this.state,
+				...partialState
+			};
+
+			const value = `${formatHours(stateValues[HOURS], stateValues[MERIDIAN])}:${formatDigits(stateValues[MINUTES])}`;
+			this.props.onChange(value);
+		}
```

Comment:
nitpick, you can unindent the body of this function by bailing early `if (!this.props.onChange) { return; }`